examples/with-remix-thirdpartyemailpassword/app/routes/_index.tsx

```diff
@@ -0,0 +1,110 @@
+import { CelebrateIcon, SeparatorLine, BlogsIcon, GuideIcon, SignOutIcon } from "../../assets/images";
+import { useSessionContext } from "supertokens-auth-react/recipe/session/index.js";
+import { signOut } from "supertokens-auth-react/recipe/thirdpartyemailpassword/index.js";
+import { recipeDetails } from "../config/frontend";
+import SuperTokens from "supertokens-auth-react";
+
+export default function Home() {
```

Comment:
does remix not have a concept of server side rendering?

Reply:
It does, but here's the current challenge. I can fetch and render session data on the client side with no problem using ` useSessionContext()`.

The problem is that NextJS app uses a function called `getSSRSession` that is _NextJS specific_ to get session details on the server-side, so there is currently no Remix equivalent and one probably needs to be made for fetching session data on the server to work. Even if we were to repurpose the function somehow, it accepts cookies and headers provided by `next/headers` with no equivalent and Remix. So objects need to be manually constructed which requires extensive troubleshooting.

Reply:
We dont need to create anythign specific to remix in our SDK. The exported custom framework allows to integrate with any framework. For example, initially, we didn't have the nextjs specific functions in our sdk, and all in the docs - which worked just fine.

From a supertokens' point of view, you want to [call getSession](https://github.com/supertokens/supertokens-node/blob/master/lib/ts/nextjs.ts#L177) which accepts a baseRequest (type is PreParsedRequest), and baseResponse (type is CollectingResponse). Now how those are formed can be completely different for remix. You do not have to copy nextjs. In nextjs, the way you get cookies during SSR is via the `cookies` function from their framework, im pretty sure remix has some way to get cookies during SSR - try googling "get cookies during SSR in remix"

Reply:
Done. Session data is now fetched on the server-side.